code/game/objects/effects/abnormality.dm

```diff
@@ -37,3 +37,77 @@
 /obj/effect/qoh_sygil/proc/fade_out()
 	animate(src, alpha = 0, time = 1 SECONDS)
 	QDEL_IN(src, 1 SECONDS)
+
+/obj/effect/magic_bullet
+	name = "magic bullet"
+	desc = "A black bullet wreathed in blue-white energy, screaming forth at an unfathomable speed."
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "magicbullet"
+	throwforce = 100
+	move_force = INFINITY
+	move_resist = INFINITY
+	pull_force = INFINITY
+	density = TRUE
+	anchored = TRUE
+	generic_canpass = FALSE
+	movement_type = PHASING | FLYING
+	var/already_moved = 0
+
+/obj/effect/magic_bullet/Initialize()
+	..()
+	playsound(get_turf(src), 'sound/abnormalities/freischutz/shoot.ogg', 100, 1)
+
+/obj/effect/magic_bullet/Moved() // Shamelessly stolen code from immovable rod and paradise lost, GO!
+	var/list/nearmiss = list()
+	if(dir == (EAST||WEST))
+		nearmiss += get_step(src, SOUTH).contents
+		nearmiss += get_step(src, NORTH).contents
+	else
+		nearmiss += get_step(src, EAST).contents
+		nearmiss += get_step(src, WEST).contents
+	for(var/mob/living/L in loc.contents)
+		L.apply_damage(120, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+		visible_message("<span class='boldwarning'>[src] pierces through [L]!</span>")
+		to_chat(L, "<span class='userdanger'>[src] slams through you!</span>")
+	for(var/mob/living/L in nearmiss)
+		L.apply_damage(60, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+		visible_message("<span class='boldwarning'>[src] just barely brushes past [L]!</span>")
+		to_chat(L, "<span class='userdanger'>[src] grazes your side!</span>")
+	if((src.x > 245) || (src.x < 10) || (src.y > 245) || (src.y < 10))
+		qdel(src)
+		return
+	var/obj/effect/frei_trail/trayl = new(src.loc)
+	trayl.dir = src.dir
+	src.forceMove(get_step(src,src.dir))
+	del(nearmiss)
```

Comment:
Do not use del, at least that's what I've been told, use qdel instead.

Reply:
Oh, you don't have to delete the list at all, it's rather pointless.

Reply:
resolved now!